packages/p2p-v2/src/pages/orders/screens/OrderDetails/OrderDetails.tsx

```diff
@@ -1,28 +1,108 @@
-import React from 'react';
-import { useLocation } from 'react-router-dom';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { FullPageMobileWrapper, PageReturn } from '@/components';
 import { useExtendedOrderDetails } from '@/hooks';
 import { p2p, useActiveAccount, useServerTime } from '@deriv/api-v2';
+import { Button, InlineMessage, Loader, Text, useDevice } from '@deriv-com/ui';
+import ChatIcon from '../../../../public/ic-chat.svg';
+import { OrderDetailsCard } from '../../components/OrderDetailsCard';
+import { OrderDetailsCardFooter } from '../../components/OrderDetailsCard/OrderDetailsCardFooter';
 import { OrdersChatSection } from '../OrdersChatSection';
+import { OrderDetailsProvider } from './OrderDetailsProvider';
+import './OrderDetails.scss';
 
-const OrderDetails = () => {
-    const location = useLocation();
-    const urlParams = new URLSearchParams(location.search);
-    const orderId = urlParams.get('order') ?? '';
-    const { data: orderInfo } = p2p.order.useGet(orderId);
+type TOrderDetailsProps = {
+    orderId: string;
+};
+
+const OrderDetails = ({ orderId }: TOrderDetailsProps) => {
+    const [showChat, setShowChat] = useState(false);
+
+    const history = useHistory();
+    const { data: orderInfo, failureReason, isError, isLoading } = p2p.order.useGet(orderId);
     const { data: activeAccount } = useActiveAccount();
     const { data: serverTime } = useServerTime();
     const { data: orderDetails } = useExtendedOrderDetails({
         loginId: activeAccount?.loginid,
         orderDetails: orderInfo,
         serverTime,
     });
+    const { isBuyOrderForUser, shouldShowLostFundsBanner } = orderDetails;
+    const { isMobile } = useDevice();
+    const headerText = `${isBuyOrderForUser ? 'Buy' : 'Sell'} USD order`;
+    const onReturn = () => history.goBack();
+    const warningMessage = 'Don’t risk your funds with cash transactions. Use bank transfers or e-wallets instead.';
+
+    if (isLoading) return <Loader isFullScreen />;
+
+    // TODO: replace with proper error screen once design is ready
+    if (isError) return <Text>{failureReason?.error.message}</Text>;
+
+    if (isMobile) {
+        if (showChat) {
```

Comment:
in mobile view, dont we directly go to the chat section when clicking on the chat icon from the table, will this handle that case as well?

Reply:
yuppp it handles this case already 👍 when we click from OrdersTable, it doesn't render OrderDetails, but it renders the Chat section

https://github.com/binary-com/deriv-app/assets/103412909/a971c006-8ffd-456e-913f-b11de203d297

Reply:
ah yeah ill fix it

Reply:
but here it is going to the order details page instead of the chat section